kusto/kcsb.go

```diff
@@ -0,0 +1,354 @@
+
+	}
+
+	return kcsb
+}
+
+//Creates a KustoConnection string builder that will authenticate with AAD user name and password.
+func (kcsb connectionStringBuilder) WithAadUserPassAuth(uname string, pswrd string, authorityID string) connectionStringBuilder {
+	requireNonEmpty(dataSource, kcsb.dataSource)
+	requireNonEmpty(aadUserId, uname)
+	requireNonEmpty(password, pswrd)
+	kcsb.aadUserID = uname
+	kcsb.password = pswrd
+	kcsb.authorityId = authorityID
+	return kcsb
+}
+
+//Creates a KustoConnection string builder that will authenticate with AAD user token
+func (kcsb connectionStringBuilder) WitAadUserToken(userTkn string) connectionStringBuilder {
+	requireNonEmpty(dataSource, kcsb.dataSource)
+	requireNonEmpty(userToken, userTkn)
+	kcsb.userToken = userTkn
+	return kcsb
+}
+
+//Creates a KustoConnection string builder that will authenticate with AAD application and key.
+func (kcsb connectionStringBuilder) WithAadAppKey(appId string, appKey string, authorityID string) connectionStringBuilder {
+	requireNonEmpty(dataSource, kcsb.dataSource)
+	requireNonEmpty(applicationClientId, appId)
+	requireNonEmpty(applicationKey, appKey)
+	requireNonEmpty(authorityId, authorityID)
+	kcsb.applicationClientId = appId
+	kcsb.applicationKey = appKey
+	kcsb.authorityId = authorityID
+	return kcsb
+}
+
+//Creates a KustoConnection string builder that will authenticate with AAD application using a certificate.
+func (kcsb connectionStringBuilder) WithAppCertificate(appId string, certificate string, thumprint string, sendCertChain bool, authorityID string) connectionStringBuilder {
+	requireNonEmpty(dataSource, kcsb.dataSource)
+	requireNonEmpty(applicationCertificate, certificate)
+	requireNonEmpty(authorityId, authorityID)
+	kcsb.applicationClientId = appId
+	kcsb.authorityId = authorityID
+
+	kcsb.applicationCertificate = certificate
+	kcsb.applicationCertificateThumbprint = thumprint
+	kcsb.sendCertificateChain = sendCertChain
+	return kcsb
+}
+
+// Creates a KustoConnection string builder that will authenticate with AAD application and an application token.
+func (kcsb connectionStringBuilder) WithApplicationToken(appId string, appToken string) connectionStringBuilder {
+	requireNonEmpty(dataSource, kcsb.dataSource)
+	requireNonEmpty(applicationToken, appToken)
+	kcsb.applicationToken = appToken
+	return kcsb
+}
+
+// Creates a KustoConnection string builder that will use existing authenticated az cli profile password.
+func (kcsb connectionStringBuilder) WithAzCli() connectionStringBuilder {
+	requireNonEmpty(dataSource, kcsb.dataSource)
+	kcsb.azcli = true
+	return kcsb
+}
+
+/*
+Creates a KustoConnection string builder that will authenticate with AAD application, using
+  an application token obtained from a Microsoft Service Identity endpoint. An optional user
+  assigned application ID can be added to the token.
+*/
+func (kcsb connectionStringBuilder) WithManagedServiceID(clientID string, resId string) connectionStringBuilder {
```

Comment:
Some pointers here since python is a little out of date.
1) only client id is needed - res id is not supported
2) if client id is not provided to Az identity it does system assigned auth
3) we prefer to have to API to avoid confusion - With System Managed Identity & With User Assigned Managed Identity

Reply:
![image](https://user-images.githubusercontent.com/102726227/193092193-91bda271-63a8-46f8-81b7-5919a174dd25.png)
Sure @yogilad ! but would like to mention the point that `azidentity` have this provision to provide `resourceId` for `managedId`. If the plan is to not support `resourseId`, then it can be removed.

Reply:
Remove resourceId entirely, we don't support it.
And like yochai said, make a method
`WithUserManagedIdentity` that accepts a clientID
And 
`WithSystemManagedIdentity` that doesn't